.github/actions/setup-windows-builder/action.yaml

```
@@ -38,8 +38,7 @@ runs:
     - name: Setup Rust toolchain
       shell: bash
       run: |
-        rustup update stable
-        rustup toolchain install stable
+        rustup toolchain install stable --no-self-update
```

I think it helps to give some context about the rationale for certain changes (so 
they are less likely to get lost in refactors):

```suggestion
        # Avoid self update to avoid CI failures: https://github.com/apache/arrow-datafusion/issues/9653
        rustup toolchain install stable --no-self-update
```